core/trino-main/src/main/java/io/trino/operator/aggregation/AggregationMaskCompiler.java

```diff
@@ -175,22 +175,23 @@ public static Constructor<? extends AggregationMaskBuilder> generateAggregationM
         Variable selectedPositionsIndex = scope.declareVariable("selectedPositionsIndex", body, constantInt(0));
         Variable rawIds = scope.declareVariable(int[].class, "rawIds");
         Variable rawIdsOffset = scope.declareVariable(int.class, "rawIdsOffset");
+        Variable dictionaryBlockPosition = scope.declareVariable("dictionaryBlockPosition", body, constantInt(0));
         body.append(new IfStatement()
                 .condition(maskBlock.instanceOf(DictionaryBlock.class))
                         .ifTrue(new BytecodeBlock()
-                                .append(maskValueBlock.set(maskBlock.cast(DictionaryBlock.class).invoke("getDictionary", ValueBlock.class, position).cast(ByteArrayBlock.class)))
```

Comment:
Would you mind putting somewhere how the generated code changes (before/after)? Can be a commit message or PR description. This would help a lot in understanding this change.

Reply:
i can pretend i understood the change, but i would prefer not to